src/ontodia/diagram/view.ts

```diff
@@ -417,19 +417,19 @@ export class DiagramView extends Backbone.Model {
         }
     }
 
-    getLinkStyle(linkTypeId: string): LinkStyle {
+    getLinkStyle(link: LinkModel): LinkStyle {
         let style = getDefaultLinkStyle();
         for (const resolver of this.linkStyleResolvers) {
-            const result = resolver(linkTypeId);
+            const result = resolver(link);
             if (result) {
                 merge(style, cloneDeep(result));
                 break;
             }
         }
-        if (!this.linkMarkers[linkTypeId]) {
-            this.linkMarkers[linkTypeId] = {
-                start: this.createLinkMarker(linkTypeId, true, style.markerSource),
-                end: this.createLinkMarker(linkTypeId, false, style.markerTarget),
+        if (!this.linkMarkers[link.linkTypeId]) {
+            this.linkMarkers[link.linkTypeId] = {
```

Comment:
This change will cause an issue in case when there are multiple link styles with the same `linkTypeId` defined.